### DISCUSSION TITLE
请问在 5.0 中该如何替换 Ant Design 自带的样式？

### DISCUSSION THREAD

**Discussion by enzeberg** [ant-design/ant-design#38740] [Q&A]
在 v4 中， 可以比较方便地做到这一点。比如，想要改变 Header 的背景色，可以在 `craco.config.js` 中改变 `@layout-header-background` 这个样式变量。

但在 v5 中，我好像没有找到这种替换样式的方法。我知道可以用 _行内样式_ 来改变 Header 的背景色，但感觉那种做法不优雅，且是一种浪费，因为浏览器加载了两份样式，但其中一份却被 _行内样式_ 顶替了。

所以想请问一下，在 v5 中，该如何 **替换** Ant Design 自带的样式？

**Comment by MadCcc** [comment]
```
<ConfigProvider theme={{ components: { Layout: { colorBgHeader: '你希望的颜色' } } }}></ConfigProvider>
```
我们后续会将 **确定的** 组件 token 加入到组件文档中。

**Reply by afc163** [reply]
这个『确定的』很辛酸呐 [狗头]

**Reply by enzeberg** [reply]
感谢！不过我试了一下，`colorBgHeader` 好像不奏效，背景色仍然是自带的，`colorPrimary` 是可以的。

```
<ConfigProvider
  theme={{
    token: {
      colorPrimary: 'orange',
    },
    components: {
      Layout: {
        colorBgHeader: 'white'
      }
    }
  }}
>
  <Header style={{ height: 'auto', }}></Header>
</ConfigProvider>
```

**Reply by wssll789** [reply]
> ```
> <ConfigProvider theme={{ components: { Layout: { colorBgHeader: '你希望的颜色' } } }}></ConfigProvider>
> ```
> 
> 我们后续会将 **确定的** 组件 token 加入到组件文档中。

这个方式可以修改 layout的header背景色，但是还有很多组件不能用这种方式修改，比如Segmented的bgColorSelected无效

**Comment by Me-UserName** [comment]
Button 在 type='link'时候，自定义colorPrimary 不生效
